### PR TITLE
Replaced nms with ProtocolLib which should fix workflows

### LINKED ISSUE #21: Gradle workflows don't work
### Expected behavior

The workflow should be able to build the plugin jar automatically for every new commit made in `master` branch and pull requests.

### Actual behavior

It fails because it requires Build Tools.

### Checks

- [X] I asked in the Discord to see whether this issue is in fact a bug that needs to be fixed.

### Anything else

I don't want to require/use Build Tools.
This can be done by replacing anything that imports `net.minecraft` with something like [Protocol Library](https://www.spigotmc.org/resources/1997).

### PR DESCRIPTION
**Pull request by Lianecx**
Spigot API 1.17 is used because i cannot build it with api v1.17.1 for some reason. This shouldnt cause any errors but if you really want me to update to 1.17.1 i can do so (the build error most likely doesnt happen for you).

Plugin has been tested on 1.17.1 and 1.18.1

**The build still has to be run with `gradle shadowJar`**

fixes #21

### CODE REVIEW

**Comment by Lianecx** [comment]
Just noticed: kind of weird that it included the commit from my gradle pr

**Comment by Lianecx** [comment]
Checks passed :)

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -2,37 +2,30 @@
...
 name: Java CI with Gradle
-on:
-  push:
-    branches: [ master ]
-    paths:
-      - 'src/**'
-#  pull_request:
-#    paths:
-#      - 'src/**'
+on: [pull_request]
```
Comment (srnyx) [inline .github/workflows/gradle.yml]: Change this back.
The current method runs on push only when a file inside of the `src` folder is edited. This prevents the action from running when something like a GitHub file (example: this file) is changed.

File: .github/workflows/gradle_pr.yml [hunk 2]
```diff
@@ -2,38 +2,30 @@
...
-on:
-#  push:
-#    branches: [ master ]
-#    paths:
-#      - 'src/**'
-  pull_request:
-#    branches: [ master ]
-    paths:
-      - 'src/**'
+on: [pull_request]
```
Comment (srnyx) [inline .github/workflows/gradle_pr.yml]: Change this back as well

File: build.gradle [hunk 3]
```diff
@@ -45,6 +45,11 @@ shadowJar {
     }
 }
 
+jar {
+    dependsOn("shadowJar")
+    enabled = false
+}
+
```
Comment (srnyx) [inline build.gradle:52]: https://github.com/Iridium-Development/IridiumSkyblock/blob/accb95c813029bbca9ae2f4b6eddc06b9aec562e/build.gradle.kts is written in Kotlin and this `jar` thing is under `tasks` with a bunch of other stuff. Look into it a bit more.

Try adding all of this: https://github.com/Iridium-Development/IridiumSkyblock/blob/accb95c813029bbca9ae2f4b6eddc06b9aec562e/build.gradle.kts#L60-L110
*But remove obvious stuff like https://github.com/Iridium-Development/IridiumSkyblock/blob/accb95c813029bbca9ae2f4b6eddc06b9aec562e/build.gradle.kts#L73-L76 and change stuff like https://github.com/Iridium-Development/IridiumSkyblock/blob/accb95c813029bbca9ae2f4b6eddc06b9aec562e/build.gradle.kts#L68*

File: .github/workflows/gradle_pr.yml [hunk 4]
```diff
@@ -2,8 +2,7 @@
 # For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
 
 name: Java CI with Gradle [PRs]
-on: pull_request
-  workflow_dispatch:
+on: [pull_request, workflow_dispatch]
```
Comment (srnyx) [inline .github/workflows/gradle_pr.yml]: No, change it entirely back to this:
```yml
on:
#  push:
#    branches: [ master ]
#    paths:
#      - 'src/**'
  pull_request:
#    branches: [ master ]
    paths:
      - 'src/**'
```

File: .github/workflows/gradle.yml [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
 
 name: Java CI with Gradle
-on: [pull_request]
+on: push
```
Comment (srnyx) [inline .github/workflows/gradle.yml]:
```yml
on:
  push:
    branches: [ master ]
    paths:
      - 'src/**'
#  pull_request:
#    paths:
#      - 'src/**'
```

**Comment by Lianecx** [comment]
Okay workflow works now

**Review by srnyx — APPROVED** [review]
Gradle stuff looks good. I don't understand any of the ProtocolLib stuff so @commandgeek will have to review that.

**Review by commandgeek — APPROVED** [review]

**Pull request merged**